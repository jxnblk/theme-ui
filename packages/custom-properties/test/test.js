import React from 'react'
import renderer from 'react-test-renderer'
import toCustomProperties, { withCustomProperties } from '../src'

const theme = {
  colors: {
    text: '#000',
    background: '#fff',
    primary: '#07c',
    secondary: '#05a',
    accent: '#609',
    muted: '#f6f6f6',
    dark: {
      foreground: {
        text: '#000',
      },
      background: {
        surface: '#fff',
      },
    },
  },
  fonts: {
    body:
      'system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", sans-serif',
    heading:
      'system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", sans-serif',
    monospace: 'Menlo, monospace',
  },
  fontWeights: {
    body: 400,
    heading: 700,
    bold: 700,
  },
  lineHeights: [1.5, 1.125],
  space: [0, 2, 3, 4, 5, 6],
  size: ['10em', '20em', '30em', '40em'],
}

it('transforms a theme config to CSS custom properties', () => {
  const result = toCustomProperties(theme)

  expect(result).toMatchSnapshot()
})

it('transforms a theme config to CSS custom properties with prefix', () => {
  const result = toCustomProperties(theme, '🍭')

  expect(result).toMatchSnapshot()
})

it('adds the default className on the wrapping element of ThemeProvider', () => {
  const ExtendedThemeProvider = withCustomProperties('🍭')
  const themeProvider = renderer.create(
    <ExtendedThemeProvider theme={theme}>
      <p> Hello world! </p>
    </ExtendedThemeProvider>
  )

  let tree = themeProvider.toJSON()
  expect(tree).toMatchSnapshot()
})

it('adds a specific className on the wrapping element of ThemeProvider', () => {
  const ExtendedThemeProvider = withCustomProperties('🍭', '🍧')
  const themeProvider = renderer.create(
    <ExtendedThemeProvider theme={theme}>
      <p> Hello world! </p>
    </ExtendedThemeProvider>
  )

  let tree = themeProvider.toJSON()
  expect(tree).toMatchSnapshot()
})
