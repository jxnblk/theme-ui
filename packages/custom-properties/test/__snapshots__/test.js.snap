// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds a specific className on the wrapping element of ThemeProvider 1`] = `
<div
  className="🍧"
>
  <p>
     Hello world! 
  </p>
</div>
`;

exports[`adds the default className on the wrapping element of ThemeProvider 1`] = `
<div
  className="theme-ui-provider"
>
  <p>
     Hello world! 
  </p>
</div>
`;

exports[`transforms a theme config to CSS custom properties 1`] = `
Object {
  "--color-accent": "#609",
  "--color-background": "#fff",
  "--color-dark-background-surface": "#fff",
  "--color-dark-foreground-text": "#000",
  "--color-muted": "#f6f6f6",
  "--color-primary": "#07c",
  "--color-secondary": "#05a",
  "--color-text": "#000",
  "--font-body": "system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", sans-serif",
  "--font-heading": "system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", sans-serif",
  "--font-monospace": "Menlo, monospace",
  "--fontWeight-body": 400,
  "--fontWeight-bold": 700,
  "--fontWeight-heading": 700,
  "--lineHeight-0": 1.5,
  "--lineHeight-1": 1.125,
  "--size-0": "10em",
  "--size-1": "20em",
  "--size-2": "30em",
  "--size-3": "40em",
  "--space-0": 0,
  "--space-1": 2,
  "--space-2": 3,
  "--space-3": 4,
  "--space-4": 5,
  "--space-5": 6,
}
`;

exports[`transforms a theme config to CSS custom properties with prefix 1`] = `
Object {
  "--🍭-color-accent": "#609",
  "--🍭-color-background": "#fff",
  "--🍭-color-dark-background-surface": "#fff",
  "--🍭-color-dark-foreground-text": "#000",
  "--🍭-color-muted": "#f6f6f6",
  "--🍭-color-primary": "#07c",
  "--🍭-color-secondary": "#05a",
  "--🍭-color-text": "#000",
  "--🍭-font-body": "system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", sans-serif",
  "--🍭-font-heading": "system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, \\"Helvetica Neue\\", sans-serif",
  "--🍭-font-monospace": "Menlo, monospace",
  "--🍭-fontWeight-body": 400,
  "--🍭-fontWeight-bold": 700,
  "--🍭-fontWeight-heading": 700,
  "--🍭-lineHeight-0": 1.5,
  "--🍭-lineHeight-1": 1.125,
  "--🍭-size-0": "10em",
  "--🍭-size-1": "20em",
  "--🍭-size-2": "30em",
  "--🍭-size-3": "40em",
  "--🍭-space-0": 0,
  "--🍭-space-1": 2,
  "--🍭-space-2": 3,
  "--🍭-space-3": 4,
  "--🍭-space-4": 5,
  "--🍭-space-5": 6,
}
`;
